### PR TITLE
Fixes AnimatedImageView designated inilializer not work

### LINKED ISSUE #497: PINAnimatedImageView Initializer with GIF
Right now, if you initialize an instance of an `PINAnimatedImageView` using its initializer that takes in a `PINCachedAnimatedImage` it will not load the image due to the way the `commonInit` works:

```
- (void)commonInit:(PINCachedAnimatedImage *)animatedImage
{
    _animatedImage = animatedImage;
    _animatedImageRunLoopMode = NSRunLoopCommonModes;
}
```

It sets the iVar, but for the animated image to show you need:

`- (void)setAnimatedImage:(PINCachedAnimatedImage *)animatedImage`

**TL;DR:** If you've got an animated image and don't want to use the category to download it from a URL, and instead want to use it's initializer you've got to do this:

```
myAnimatedImageView = [PINAnimatedImageView new];
[myAnimatedImageView setAnimatedImage:theGIF];
```
when you should be able to do this:
```
myAnimatedImageView =  [[PINAnimatedImageView alloc] initWithAnimatedImage:theGIF];
```

Locally, I've verified that if you use `animatedImage`'s setter, it works fine using the initializer:
```
- (void)commonInit:(PINCachedAnimatedImage *)animatedImage
{
    self.animatedImage = animatedImage;
    _animatedImageRunLoopMode = NSRunLoopCommonModes;
}
```

### PR DESCRIPTION
**Pull request by zhongwuzw**
Fixes #497 .

### CODE REVIEW

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by bolsinga** [comment]
Please add a test.

File: Source/Classes/AnimatedImages/PINAnimatedImageView.m [hunk 1]
```diff
@@ -61,6 +61,31 @@ - (void)commonInit:(PINCachedAnimatedImage *)animatedImage
 {
     _animatedImage = animatedImage;
     _animatedImageRunLoopMode = NSRunLoopCommonModes;
+    
+    [self initializeAnimatedImage:animatedImage];
```
Comment (garrettmoon) [inline Source/Classes/AnimatedImages/PINAnimatedImageView.m]: Can you add a nil check here and make mark `initializeAnimatedImage:` as taking a non null animatedImage?

**Comment by garrettmoon** [comment]
Looks like the right fix to me! @DreamingInBinary do you mind testing to make sure this addresses your issue?

**Comment by zhongwuzw** [comment]
Added test.

**Comment by bolsinga** [comment]
Thanks for adding a test!

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by garrettmoon** [comment]
Mind adding an entry to CHANGELOG.md for this one?

**Comment by pinterest-ios** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Any source code changes should have an entry in CHANGELOG.md or have #trivial in their title.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by zhongwuzw** [comment]
Changelog added, I'll add changelog for all my PRs. :)

**Comment by pinterest-ios** [comment]
🚫 CI failed with [log](https://gist.githubusercontent.com/pinterest-ios/3e279ec162d62fd44d09aba747f812fd/raw/8866ef8fd82c56df335dda7a5d0c3d6e446498bd/log.txt)

**Comment by garrettmoon** [comment]
Mind rebasing against master? I think I fixed a flakey test.

**Pull request merged**